src/apps/config/alarm_codec.lua

```diff
@@ -120,7 +134,7 @@ local function normalize (t, attrs)
    t = t or {}
    local ret = {}
    for _, k in ipairs(attrs) do
-      table.insert(ret, t[k] or '')
+      table.insert(ret, t[k])
```

Comment:
Does this do the right thing if `t[k]` is nil?  Would seem to me like `#ret` would stay the same.

Reply:
The goal of this function is to decompress a sparse table into an list of values in a specific order. For instance in the call:

```lua
alarm:raise(key, {alarm_text='text'})
```

The function decompresses `args` ({alarm_text='text'}) to arguments `perceived_severity` and `alarm_text` (which are stored in ret). In this case `args` doesn't define `perceived_severity` so `ret` will contain {nil, 'text'}. The size of `ret` is always the same regardless of the size of #t (sometimes might be the same, but decompressing the list in a specific order is important too).

Reply:
```
wingo@sparrow ~/src/snabb/src$ cat /tmp/foo.lua
local x = {}
print(#x)
table.insert(x, nil)
print(#x)
wingo@sparrow ~/src/snabb/src$ sudo ./snabb snsh /tmp/foo.lua
0
0
```

Are you sure it would end up being `{nil, 'text'}` ?  Looks to me like it might be `{'text'}`.

Reply:
Yep, you're right. It doesn't work as I expected. Inserting a nil in a table has no effect.

If I leave an empty string as a  placeholder for nil values, all arguments will be encoded as strings. OTOH, I can use something else as a placeholder and make `maybe_string` to consider that placeholder a nil value. WDYT?

Reply:
If your intention is really to unroll some possibly-nil positional arguments, then perhaps use:
```
local args = {}
for i,k in ipairs(attrs) do args[i] = t[k] done
```